core/templates/domain/question/question-backend-api.service.ts

```diff
@@ -204,6 +222,39 @@ export class QuestionBackendApiService {
       this._fetchQuestionSummaries(skillId, offset, resolve, reject);
     });
   }
+
+  async fetchDiagnosticTestQuestionsAsync(
+      topicId: string
+  ): Promise<SkillIdToQuestionsResponse> {
+    return new Promise((resolve, reject) => {
+      const diagnosticTestQuestionsURL = (
+        this.urlInterpolationService.interpolateUrl(
+          '/diagnostic_test_questions_handler_url/<topic_id>', {
+            topic_id: topicId
+          }));
+
+      this.http.get<SkillIdToQuestionsBackendResponse>(
+        diagnosticTestQuestionsURL).toPromise().then((response) => {
+        let skillIdToQuestionsDict: SkillIdToQuestionsResponse = {};
+
+        for (let skillId in response.skill_id_to_questions_dict) {
+          skillIdToQuestionsDict[skillId] = {};
```

Comment:
Don't use raw dicts, you should have a model domain object for this with specified keys (see .model.ts files).

Reply:
Created